### PR TITLE
syntax-highlighter,symbols: support Apex

### PR DESCRIPTION
**Pull request by tjdevries**
Update syntect for Apex.

I think I should run `./publish` here as well?

### CODE REVIEW

**Comment by tjdevries** [comment]
reference: https://github.com/sourcegraph/sourcegraph/issues/25238

**Review by slimsag — APPROVED** [review]

**Comment by slimsag** [comment]
looks fine, please make sure to test that it actually works on apex files and add a CHANGELOG entry for it once you have. Sometimes syntax files are incompatible and will not work.

You shouldn't need to run any `publish` script anymore, since Dave merged the code into this repo. But you're the first one trying this, so there may be issues lurking :) I think CI will publish it for you tho.

You will need to update these SHA references once it is published: https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+sourcegraph/syntect_server:32d88&patternType=literal

**Comment by tjdevries** [comment]
When I run `./build.sh` I get:

```
Successfully built 27bb191f8398
Successfully tagged sourcegraph/syntax-highlighter:latest
```

So that seems good.

@slimsag should I also run `./publish.sh` in this PR? or should it be done in a separate PR after this is merged? I think this is the first PR like this since this has been moved into sg/sg

**Comment by slimsag** [comment]
delete all mentions of a publish.sh script for me please :)

**Comment by tjdevries** [comment]
Ah, got it for `./publish` (didn't see your message). Will look at the rest now. Thanks :)

File: docker-images/syntax-highlighter/rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-1.46.0
+1.53.0
```
Comment (chrismwendt) [inline docker-images/syntax-highlighter/rust-toolchain:1]: Just a heads up: I bumped rust to 1.53.0 so it works on my Mac M1

Comment (slimsag) [reply docker-images/syntax-highlighter/rust-toolchain:1]: cool, as long as it builds still that should be no problem.

Comment (chrismwendt) [reply docker-images/syntax-highlighter/rust-toolchain:1]: Confirmed it still builds and runs on an x86_64 GCP E2 machine ✅

File: cmd/frontend/internal/highlight/syntect_language_map.go [hunk 2]
```diff
@@ -4,6 +4,7 @@ package highlight
 // to valid syntect file extensions.
 var SyntectLanguageMap = map[string]string{
 	"txt":                  "txt",
+	"apex":                 "apex",
```
Comment (tjdevries) [inline cmd/frontend/internal/highlight/syntect_language_map.go:7]: are there any other filetypes we should add here?

File: docker-images/syntax-highlighter/src/main.rs [hunk 3]
```diff
@@ -106,14 +106,50 @@ fn highlight(q: Query) -> JsonValue {
...
-                .or_else(|| syntax_set.find_syntax_by_extension(extension))
-                .or_else(|| syntax_set.find_syntax_by_first_line(&q.code))
-                .unwrap_or_else(|| {
-                    is_plaintext = true;
-                    syntax_set.find_syntax_plain_text()
-                })
+
+            // Override syntect's language detection for conflicting file extensions because
+            // it's impossible to express this logic in a syntax definition.
+            struct Override {
```
Comment (tjdevries) [inline docker-images/syntax-highlighter/src/main.rs:112]: maybe add a unit test for this?

File: cmd/frontend/internal/highlight/syntect_language_map.go (hunk 2 above)
Comment (Strum355) [reply cmd/frontend/internal/highlight/syntect_language_map.go:7]: Maybe the ones listed in the salesforce vscode extension? https://sourcegraph.com/github.com/forcedotcom/salesforcedx-vscode@develop/-/blob/packages/salesforcedx-vscode-apex/package.json?L328-L332

Comment (chrismwendt) [reply cmd/frontend/internal/highlight/syntect_language_map.go:7]: This list is for markdown code fences. Will anyone use anything other than `apex` to mean Apex?

Comment (tjdevries) [reply cmd/frontend/internal/highlight/syntect_language_map.go:7]: ah, ignore this part then :)

File: docker-images/syntax-highlighter/src/main.rs (hunk 3 above)
Comment (chrismwendt) [reply docker-images/syntax-highlighter/src/main.rs]: Added a test for determining the language of different `.cls` files.

**Review by chrismwendt — APPROVED** [review]
Me approving a PR that I wrote half of:

CleanShot 2021-09-23 at 17 21 42@2x

**Comment by tjdevries** [comment]
This seems good to me, probably should just check and make sure that all the commit hashes are right since we have so many repos and commits :laughing:

**Comment by chrismwendt** [comment]
Double checked the hashes ✅

**Pull request merged**